### PR TITLE
[Feature] Add dependencies_required and api_keys_required

### LINKED ISSUE #392: [Feature Request] Add `dependencies_required` and `api_keys_required`
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

There are many places we throw import error or api key missing error. We should introduce decorators such as `dependencies_required` and `api_keys_required` to handle different dependencies or api keys missing error.

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ArnoldIOI**
## Description

In this update, two new decorators, `dependencies_required` and `api_keys_required`, have been implemented to streamline the error handling for different dependencies or api keys missing error.

## Motivation and Context
`close #392`
[link to the issue here.](https://github.com/camel-ai/camel/issues/392)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [x] implement `dependencies_required`
- [X] implement `api_keys_required`
- [X] add test case for `dependencies_required`
- [X] add test case for `api_keys_required`

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [X] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [X] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/utils/commons.py [hunk 1]
```diff
@@ -241,3 +242,93 @@ def check_server_running(server_url: str) -> bool:
...
+    Callable[[F], F]: The original function with the added check for
+    required module dependencies.
+
+    Raises:
+    ImportError: If any of the required modules are not available.
+
+    Example:
+    @dependencies_required('numpy', 'pandas')
+    def data_processing_function():
+        # Function implementation...
```
Comment (dandansamax) [inline camel/utils/commons.py:265]: Our project usually uses google style docstring. Could you please fix them?

Here is an example:

```suggestion
    Args:
        required_modules (str): The required modules to be checked for
            availability.

    Returns:
        Callable[[F], F]: The original function with the added check for
            required module dependencies.

    Raises:
        ImportError: If any of the required modules are not available.

    Example:
        ::
        
            @dependencies_required('numpy', 'pandas')
            def data_processing_function():
                # Function implementation...
```

File: camel/utils/commons.py [hunk 2]
```diff
@@ -241,3 +242,93 @@ def check_server_running(server_url: str) -> bool:
...
+
+
+def is_module_available(module_name: str) -> bool:
+    r"""Check if a module is available for import.
+
+    Parameters:
+    module_name (str): The name of the module to check for availability.
+
+    Returns:
+    bool: True if the module can be imported, False otherwise.
```
Comment (dandansamax) [inline camel/utils/commons.py:292]: Same here.

File: camel/utils/commons.py [hunk 3]
```diff
@@ -241,3 +242,93 @@ def check_server_running(server_url: str) -> bool:
...
+    for required API keys.
+
+    Raises:
+    ValueError: If any of the required API keys are missing in the
+    environment variables.
+
+    Example:
+    @api_keys_required('API_KEY_1', 'API_KEY_2')
+    def some_api_function():
+        # Function implementation...
```
Comment (dandansamax) [inline camel/utils/commons.py:319]: Same here.

**Review by dandansamax — CHANGES_REQUESTED** [review]
It looks awesome! Thank you very much for these easy-to-use decorators. Just left some comments about the docstring format.

File: camel/utils/commons.py (hunk 3 above)
Comment (ArnoldIOI) [reply camel/utils/commons.py]: Thanks a lot for the clear example and explanation!
I've changed the style to follow the google docstring style.

**Comment by ArnoldIOI** [comment]
@Wendong-Fan Thanks for the review and merge. And I've updated to the latest master branch again.

**Comment by Wendong-Fan** [comment]
> @Wendong-Fan Thanks for the review and merge. And I've updated to the latest master branch again.

Hey @ArnoldIOI , the checks are failed since we are using ruff now, if you need support to fix all the errors you can reach out to me~

**Comment by ArnoldIOI** [comment]
Hey @Wendong-Fan I've updated to pass the **ruff** checks. 
In the github workflow, I noticed there're still failed checks like **openai.APIConnectionError: Connection error**. Amu hints on this?

**Review by zechengz — APPROVED** [review]

**Comment by Wendong-Fan** [comment]
> Hey @Wendong-Fan I've updated to pass the **ruff** checks. In the github workflow, I noticed there're still failed checks like **openai.APIConnectionError: Connection error**. Any hints on this?

Hey @ArnoldIOI , no worries it's just because you don't have API Key set in your environment, thanks for the contribution and I will merge the PR

**Pull request merged**